belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/InvalidRequestExceptionHandler.java

```diff
@@ -0,0 +1,217 @@
+package io.github.belgif.rest.problem.spring;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import com.atlassian.oai.validator.report.ValidationReport;
+import com.atlassian.oai.validator.springmvc.InvalidRequestException;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.github.belgif.rest.problem.BadRequestProblem;
+import io.github.belgif.rest.problem.ResourceNotFoundProblem;
+import io.github.belgif.rest.problem.api.InEnum;
+import io.github.belgif.rest.problem.api.InputValidationIssue;
+import io.github.belgif.rest.problem.api.InputValidationIssues;
+import io.github.belgif.rest.problem.api.Problem;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+/**
+ * This class is only active when Spring-Boot applications implemented the Atlassian swagger-request-validator.
+ * Maps the InvalidRequestException thrown by the swagger-request-validator to belgif problems.
+ */
+@RestControllerAdvice
+@ConditionalOnWebApplication
+@ConditionalOnClass(InvalidRequestException.class)
+@Order(1)
+// @Order(1) to take precedence over io.github.belgif.rest.problem.spring.ProblemExceptionHandler
+public class InvalidRequestExceptionHandler {
```

Comment:
Added testclasses with unit tests.
I couldn't easily mock the static methods in the utils class because adding the mockito-inline dependency breaks the other mocks.
I think this will work as well.